14/umbraco-cms/reference/security/external-login-providers.md

```diff
@@ -512,14 +512,96 @@ export default MyLitView;
 
 {% endtabs %}
 
+### Generic backoffice login provider composer
+A composer and `genericAuthenticationOptions` configuration class to setup the authentication options for the generic authentication provider using dependency injection.
+
+{% code title="GenericBackOfficeExternalLoginComposer.cs" lineNumbers="true" %}
+```csharp
+// this example uses a non existing generic OathProvider NuGet package
+// but any package that uses the Microsoft.AspNetCore.Authentication.OAuth.OathOptions will work
+using AspNet.Security.OAuth.Generic;
+using Microsoft.Extensions.Options;
+using Umbraco.Cms.Api.Management.Security;
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Web.Common.Helpers;
+
+namespace MyUmbracoProject.CustomAuthentication;
+
+public class GenericBackOfficeExternalLoginComposer : IComposer
+{
+    public void Compose(IUmbracoBuilder builder)
+    {
+        // register the generic loginProvider options (auto linking, manual linking, ...)
+        builder.Services.ConfigureOptions<GenericBackOfficeExternalLoginProviderOptions>();
+        // register the generic authtication options (secret, callback, errorhandling, ...)
+        builder.Services.ConfigureOptions<ConfigureGenericAuthenticationOptions>();
+
+        builder.AddBackOfficeExternalLogins(logins =>
+        {
+            logins.AddBackOfficeLogin(
+                backOfficeAuthenticationBuilder =>
+                {
+                    // this Add... method will be part of the OathProvider nuget package you install
+                    backOfficeAuthenticationBuilder.AddGenericProvider(
```

Comment:
am getting the following error on AddGenericProvider here:

![image](https://github.com/umbraco/UmbracoDocs/assets/83591955/a13f5ebf-61bb-473b-8e11-5667407ef726)